src/storage/store/string_column.cpp

```diff
@@ -28,6 +28,9 @@ StringColumn::StringColumn(std::string name, std::unique_ptr<LogicalType> dataTy
 
 void StringColumn::scanOffsets(Transaction* transaction, const ReadState& state,
     string_offset_t* offsets, uint64_t index, uint64_t numValues, uint64_t dataSize) {
+    // TODO(bmwinger): this causes an issue with structs where a read is attempted on an empty chunk
```

Comment:
I don't follow here, what's the connection with structs?

Reply:
I think it was that for some reason structs are reading the first value, even if the numValues in the metadata is 0, so the assertion fails for strings stored in a struct. As long as everything is zero, that value just ends up being an empty string, but it shouldn't be being read. I'm not sure why it was just for structs.

Reply:
Do you have a test case or example that can reproduce that? It looks like we should fix it now or later, better document it down in an issue?

Reply:
Un-commenting that line, it causes only ` EmptyCreateNodeTest.CreateStructWithWriteTransaction` to fail. That said, after creating the node, the string chunk should contain a single string, so maybe it's the metadata that's wrong, something to do with the scan occurring before the commit probably.